### ISSUE TITLE
Different default colourmap for `vis.mpl`

### ISSUE THREAD

**Issue by mtb-za** [fatiando/fatiando#231]
On exploring the `vis.mpl` methods, I noticed that we are using the Jet colourmap by default. There is lots of evidence that this is not a very good choice, so I propose that this be changed to something else.

The matplotlib [docs](http://matplotlib.org/users/colormaps.html) have a good summary of the various options available. Many of these would likely be a better choice.

**Comment by mtb-za** [comment]
As an example.

This is using the RdBu colourmap, which is well-suited for diverging data:
![tdx1](https://cloud.githubusercontent.com/assets/1148889/9378721/fd3f3a78-4723-11e5-920a-4ac8ab77b2dd.png)
This is the default, using JET.
![tdx2](https://cloud.githubusercontent.com/assets/1148889/9378722/fdc69b8a-4723-11e5-8dde-1e6071c96224.png)
This is using a reversed PuBu:
![tdx3](https://cloud.githubusercontent.com/assets/1148889/9378873/36befeea-4725-11e5-9114-c8fa7b956bc4.png)

Comments? I know that there are some better rainbow-ish colourmaps available, including a possible new default for matplotlib: https://www.youtube.com/watch?v=xAoljeRJ3lU

**Comment by leouieda** [comment]
@mtb-za I completely agree that jet is terrible and should not be the default. I use the `RdBu_r` colormap for all of my plotting.

The good news is that we don't have to change anything. matplotlib is changing the default [starting with version 2.0](http://matplotlib.org/style_changes.html) in September. 

Another thing, I'm really not happy with the way the `fatiando.vis.mpl` module works. This was hack I did some 4 years ago to make the `contour`, `contourf` and `pcolor` functions behave like I wanted them to. So `mpl` imports `from matplotlib.pyplot  import *` and then overloads these functions. I did this so that I didn't need to import both `mpl` and `pyplot`. 

Turns out this is terrible because I have to repeat a lot of matplotlib code when it changes. Since then, I discovered `tricontourf` and `tripcolor` which make the interpolation functionality in `mpl.contourf` and `mpl.pcolor` completely useless (and terrible for large data). So there is no reason to keep the `mpl` module as it is now and I strongly recommend not using it as a replacement for `pyplot`. More on this transition on #204 (would love some comments there).
